client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+    const annotationNode = useRef(null);
+    useEffect(() => {
+      // Add a ref to the annotation, allowing the comment to float alongside the attached text.
+      // This adds rather than sets the ref, so that a comment may be attached across paragraphs or around entities
+      const annotation = commentApp.layout.commentAnnotations.get(commentId)
+      if (annotation) {
+        annotation.addRef(annotationNode);
+      }
+      return () => {
+        const annotation = commentApp.layout.commentAnnotations.get(commentId)
+        if (annotation) {
+          annotation.removeRef(annotationNode);
+        }
+      }
+    });
+    const onClick = () => {
+      // Pin and focus the clicked comment
+      commentApp.store.dispatch(
+        commentApp.actions.setFocusedComment(commentId)
+      );
+      commentApp.store.dispatch(
+        commentApp.actions.setPinnedComment(commentId)
+      );
+    }
+    // TODO: determine the correct way to make this accessible, allowing both editing and focus jumps
+    return (
+      <a 
+        type="button"
+        ref={annotationNode}
+        onClick={onClick}
+        data-annotation
+      >
+        {children}
+      </a>
+    )
+  }
+  return CommentDecorator
+}
+
+function forceResetEditorState(editorState, replacementContent) {
+  const content = replacementContent ? replacementContent : editorState.getCurrentContent()
+  const state = EditorState.set(
+    EditorState.createWithContent(
+      content,
+      editorState.getDecorator(),
+    ),
+    {
+      selection: editorState.getSelection(),
+      undoStack: editorState.getUndoStack(),
+      redoStack: editorState.getRedoStack(),
+    },
+  );
+  return EditorState.acceptSelection(state, state.getSelection())
+};
+
+function CommentableEditor({commentApp, fieldNode, contentPath, rawContentState, onSave, inlineStyles, editorRef, ...options}) {
+    const [editorState, setEditorState] = useState(() => createEditorStateFromRaw(rawContentState));
+    const CommentControl = useMemo(() => getCommentControl(commentApp, contentPath, fieldNode), [commentApp, contentPath, fieldNode]);
+    const commentsSelector = useMemo(() => commentApp.utils.selectCommentsForContentPathFactory(contentPath), [contentPath, commentApp]);
+    const CommentDecorator = useMemo(() => getCommentDecorator(commentApp), [commentApp])
+    const comments = useSelector(commentsSelector, shallowEqual);
+    const enabled = useSelector(commentApp.selectors.selectEnabled);
+    const focusedId = useSelector(commentApp.selectors.selectFocused);
+
+    const ids = useMemo(() => comments.map(comment => comment.localId), [comments]);
+
+    const commentStyles = useMemo(() => ids.map(id => {return {
+      type: COMMENT_STYLE_IDENTIFIER + id,
+      style: enabled ? {
+        'background-color': (focusedId !== id) ? '#01afb0' : '#007d7e'
+      } : {}
+    }}), [ids, enabled]);
```

Comment:
I find this style of inline `return` a bit hard to read.

```suggestion
    const commentStyles = useMemo(() => ids.map(id => ({
      type: `${COMMENT_STYLE_IDENTIFIER}${id}`,
      style: enabled ? {
        'background-color': (focusedId !== id) ? '#01afb0' : '#007d7e'
      } : {}
    })), [ids, enabled]);
```

Reply:
Removed, thanks!